url/resources/percent-encoding.py

```diff
@@ -0,0 +1,23 @@
+from six.moves.urllib.parse import unquote
+from wptserve.utils import isomorphic_decode
+
+# Use numeric references to let the HTML parser take care of inserting the correct code points
+# rather than trying to figure out the necessary bytes for each encoding. (The latter can be
+# especially tricky given that Python does not implement the Encoding Standard.)
+def numeric_references(input):
+    output = b""
+    for cp in input:
+        print cp
+        output += b"&#x" + format(ord(cp), b"X") + b";"
+    return output
+
+def main(request, response):
+    value = request.GET.first(b"value")
+    encoding = request.GET.first(b"encoding")
+
+    output_value = numeric_references(unquote(value).decode(b"utf-8"))
```

Comment:
I worry a bit about all the steps of smuggling the value through URL parameters. If a browser doesn't do percent-encoding correctly, then probably it will fail to get the correct `output_value`, messing up the test.

I guess the assumption is that the browser does utf-8 percent encoding correctly?

If you wanted to avoid that assumption, you could use another encoding scheme, e.g. base64 encoding or maybe `JSON.stringify`.

Reply:
Yeah, that might be a good improvement here.

Reply:
It doesn't seem like the output of `JSON.stringify` is always ASCII so maybe I'll leave this as-is for now.

Reply:
base64 with atob/btoa and Python's equivalent seems like it'd work nicely?

Reply:
How does that work given the non-ASCII input?

Reply:
Ah, right, it's indeed a bit more complicated. https://github.com/jsdom/whatwg-url/blob/d5f65c2bef5d88f873203d33289ddfe13f2eee1f/live-viewer/live-viewer.js#L135-L150 is what I was thinking, but perhaps it's not worth it, especially given that you'd have to implement it in both Python and JS.

Reply:
I made this work, but it took longer than expected due to some weirdness from WPT.